engine-rest/engine-rest/src/main/java/org/camunda/bpm/engine/rest/impl/TaskRestServiceImpl.java

```diff
@@ -66,8 +69,23 @@ else if (Hal.APPLICATION_HAL_JSON_TYPE.equals(variant.getMediaType())) {
   }
 
   public List<TaskDto> getJsonTasks(UriInfo uriInfo, Integer firstResult, Integer maxResults) {
+    // get list of tasks
     TaskQueryDto queryDto = new TaskQueryDto(getObjectMapper(), uriInfo.getQueryParameters());
-    return queryTasks(queryDto, firstResult, maxResults);
+    queryDto.setObjectMapper(getObjectMapper());
+
+    ProcessEngine engine = getProcessEngine();
+    TaskQuery query = queryDto.toQuery(engine);
+
+    List<Task> matchingTasks = executeTaskQuery(firstResult, maxResults, query);
+
+    List<TaskDto> result;
+    if (Boolean.TRUE.equals(queryDto.getWithCommentAttachmentInfo())) {
+      result = matchingTasks.stream().map(TaskWithAttachmentAndCommentDto::fromEntity).collect(Collectors.toList());
+    }
+    else {
+      result = matchingTasks.stream().map(TaskDto::fromEntity).collect(Collectors.toList());
+    }
+    return result;
   }
 
   public HalTaskList getHalTasks(UriInfo uriInfo, Integer firstResult, Integer maxResults) {
```

Comment:
📓 No action, just comment. The param is not considered here. As Hal queries are used in Tasklist where we need performance.

Reply:
Thanks @yanavasileva for your reviews. Certainly I am willing these changes to be part of 7.22 release. I am done with addressing the comments and unit/integrations tests are running fine. Please let me know if the platform/e2e tests ran well or need something to be addressed, I will push the changes accordingly at a single instance and we can wrap it up.